src/Pact/Eval.hs

```diff
@@ -291,12 +432,16 @@ reduce (TObject ps t i) =
 reduce (TBinding ps bod c i) = case c of
   BindLet -> reduceLet ps bod i
   BindSchema _ -> evalError i "Unexpected schema binding"
-reduce t@TModule {} = evalError (_tInfo t) "Module only allowed at top level"
+reduce TModule{..} =
+  case _tModuleDef of
+    Module{} -> evalError _tInfo "Module only allowed at top level"
+    Interface{} -> evalError _tInfo "Interface only allowed at top level"
 reduce t@TUse {} = evalError (_tInfo t) "Use only allowed at top level"
 reduce t@TBless {} = evalError (_tInfo t) "Bless only allowed at top level"
 reduce t@TStep {} = evalError (_tInfo t) "Step at invalid location"
 reduce TSchema {..} = TSchema _tSchemaName _tModule _tMeta <$> traverse (traverse reduce) _tFields <*> pure _tInfo
 reduce TTable {..} = TTable _tTableName _tModule _tHash <$> mapM reduce _tTableType <*> pure _tMeta <*> pure _tInfo
+reduce t@TImplements {} = unsafeReduce t 
```

Comment:
this is an error. C.f. reduce bless above.

Reply:
Could you elaborate more on this when we meet up?